dev/com.ibm.ws.microprofile.reactive.messaging.kafka/src/com/ibm/ws/microprofile/reactive/messaging/kafka/PartitionTracker.java

```diff
@@ -72,9 +74,10 @@ public boolean isClosed() {
      *
      * @param offset the record offset
      * @param leaderEpoch the record leaderEpoch
+     * @param context the context to apply around the completion of the result, if the completion stage is completed asynchronously
      * @return a CompletionStage which completes when any associated processing has been completed (e.g. when the message offset has been committed)
      */
-    public CompletionStage<Void> recordDone(long offset, Optional<Integer> leaderEpoch) {
+    public CompletionStage<Void> recordDone(long offset, Optional<Integer> leaderEpoch, RMContext context) {
```

Comment:
How is the context here being applied to the result, given we currently aren't doing anything with it.

Reply:
Here the context is unused. We don't need to apply it in this method because we return a `CompletableFuture` which is already complete.

However, this method takes the context as an argument because the method is overridden in `CommittingPartitionTracker` which returns an incomplete `CompletableFuture` and applies the context around the call to `complete()` when that's called asynchronously.